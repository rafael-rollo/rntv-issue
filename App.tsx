import React, {useState} from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  View,
  Text,
  useWindowDimensions,
} from 'react-native';

import {TabView, SceneMap} from 'react-native-tab-view';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const FirstRoute = () => (
  <View style={styles.route}>
    <Text>First route</Text>
  </View>
);

const SecondRoute = () => (
  <View style={styles.route}>
    <Text>Second route</Text>
  </View>
);

const renderScene = SceneMap({
  first: FirstRoute,
  second: SecondRoute,
});

function App(): JSX.Element {
  const layout = useWindowDimensions();
  const [index, setIndex] = useState(0);

  const [routes] = React.useState([
    {key: 'first', title: 'First'},
    {key: 'second', title: 'Second'},
  ]);

  return (
    <SafeAreaView style={[styles.container]}>
      <StatusBar barStyle={'dark-content'} backgroundColor={Colors.lighter} />
      <TabView
        navigationState={{index, routes}}
        renderScene={renderScene}
        onIndexChange={setIndex}
        initialLayout={{width: layout.width}}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.lighter,
  },
  route: {
    flex: 1,
    backgroundColor: Colors.lighter,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default App;
