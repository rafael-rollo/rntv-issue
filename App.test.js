import React from 'react';

import {render, fireEvent} from '@testing-library/react-native';
import App from './App';

test('Should present the route title on scene content', () => {
  const {toJSON, getByLabelText, getByText} = render(<App />);

  expect(toJSON()).toMatchSnapshot(); // only first scene rendered

  expect(getByText('First route')).toBeDefined();

  fireEvent.press(getByLabelText('Second'));

  expect(getByText('Second route')).toBeDefined();
});
