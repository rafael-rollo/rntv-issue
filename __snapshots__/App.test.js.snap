// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should present the route title on scene content 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "backgroundColor": "#F3F3F3",
        "flex": 1,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      style={
        {
          "backgroundColor": "#2196f3",
          "elevation": 4,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 0.5,
          "zIndex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#ffeb3b",
              "bottom": 0,
              "height": 2,
              "left": "0%",
              "position": "absolute",
              "right": 0,
              "transform": [],
              "width": "NaN%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "overflow": "scroll",
          }
        }
      >
        <RCTScrollView
          accessibilityRole="tablist"
          alwaysBounceHorizontal={false}
          automaticallyAdjustContentInsets={false}
          collapsable={false}
          contentContainerStyle={
            [
              {
                "flexDirection": "row",
                "flexWrap": "nowrap",
              },
              {
                "flex": 1,
              },
              undefined,
            ]
          }
          data={
            [
              {
                "key": "first",
                "title": "First",
              },
              {
                "key": "second",
                "title": "Second",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          initialNumToRender={10}
          keyExtractor={[Function]}
          keyboardShouldPersistTaps="handled"
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          onViewableItemsChanged={[Function]}
          overScrollMode="never"
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={16}
          scrollsToTop={false}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={[]}
          style={{}}
          viewabilityConfigCallbackPairs={
            [
              {
                "onViewableItemsChanged": [Function],
                "viewabilityConfig": undefined,
              },
            ]
          }
        >
          <View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessibilityLabel="First"
                accessibilityRole="tab"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": true,
                  }
                }
                accessibilityStates={
                  [
                    "selected",
                  ]
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                delayPressIn={0}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "width": 0,
                      },
                    ],
                  ]
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                        "minHeight": 48,
                        "padding": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <View>
                    <View
                      collapsable={false}
                      style={
                        {
                          "opacity": 0,
                        }
                      }
                    >
                      <Text
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "color": "rgba(255, 255, 255, 0.7)",
                            "margin": 4,
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        First
                      </Text>
                    </View>
                    <View
                      collapsable={false}
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "color": "rgba(255, 255, 255, 1)",
                            "margin": 4,
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        First
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onFocusCapture={[Function]}
              onLayout={[Function]}
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <View
                accessibilityLabel="Second"
                accessibilityRole="tab"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityStates={[]}
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                delayPressIn={0}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "width": 0,
                      },
                    ],
                  ]
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                        "minHeight": 48,
                        "padding": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <View>
                    <View
                      collapsable={false}
                      style={
                        {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "color": "rgba(255, 255, 255, 0.7)",
                            "margin": 4,
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        Second
                      </Text>
                    </View>
                    <View
                      collapsable={false}
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "transparent",
                            "color": "rgba(255, 255, 255, 1)",
                            "margin": 4,
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        Second
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <RNCViewPager
      collapsable={false}
      initialPage={0}
      keyboardDismissMode="on-drag"
      layout={
        {
          "height": 0,
          "width": 750,
        }
      }
      layoutDirection="ltr"
      onMoveShouldSetResponderCapture={[Function]}
      onPageScroll={[Function]}
      onPageScrollStateChanged={[Function]}
      onPageSelected={[Function]}
      scrollEnabled={true}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            [
              {
                "flex": 1,
                "overflow": "hidden",
              },
              {
                "width": 750,
              },
              [
                undefined,
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ],
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text>
              First route
            </Text>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          style={
            [
              {
                "flex": 1,
                "overflow": "hidden",
              },
              {
                "width": 750,
              },
              [
                undefined,
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ],
            ]
          }
        />
      </View>
    </RNCViewPager>
  </View>
</RCTSafeAreaView>
`;
